packages/contractkit/src/utils/signing-utils.ts

```diff
@@ -214,3 +223,41 @@ export function decodeSig(sig: any) {
     s: ethUtil.toBuffer(s) as Buffer,
   }
 }
+
+/**
+ * Attempts each recovery key to find a match
+ */
+export function recoverKeyIndex(
+  signature: Uint8Array,
+  publicKey: BigNumber,
+  hash: Uint8Array
+): number {
+  for (let i = 0; i < 4; i++) {
+    const compressed = false
+    // Force types to be Uint8Array
+    const signatureArray = new Uint8Array(signature)
+    const hashArray = new Uint8Array(hash)
+    const recoveredPublicKeyByteArr = ecdsaRecover(signatureArray, i, hashArray, compressed)
+    const publicKeyBuff = Buffer.from(recoveredPublicKeyByteArr)
+    const recoveredPublicKey = bufferToBigNumber(publicKeyBuff)
+    console.log(
+      'Recovered key: ' + recoveredPublicKey,
+      'expected',
+      publicKey,
+      publicKey.eq(recoveredPublicKey)
+    )
+    if (publicKey.eq(recoveredPublicKey)) {
+      return i
+    }
+  }
+  throw new Error('Unable to generate recovery key from signature.')
+}
+
+export function getAddressFromPublicKey(publicKey: BigNumber): Address {
```

Comment:
@tkporter I tried consolidating this to use the util `publicKeyToAddress` method; however it introduced more problems. `ethUtil.pubToAddress` method does additional cleansing that allows it to succeed under more conditions.